Crash while changing "request_setting > max_stale_age" value

**1/ Environment** 
Terraform v0.15.4 on linux_amd64 + provider registry.terraform.io/fastly/fastly v0.30.0

**2/ Issue : plugin crash while changing "request_setting > max_stale_age" value** 
```
request_setting {
        max_stale_age = 80001  # previous value is 80000
    }
```

terraform apply
```
Stack trace from the terraform-provider-fastly_v0.30.0 plugin:

panic: interface conversion: interface {} is int, not uint

goroutine 50 [running]:
github.com/fastly/terraform-provider-fastly/fastly.(*RequestSettingServiceAttributeHandler).Process(0xc00000e608, 0xc00017f780, 0x11, 0xc000580440, 0x0, 0x1)
        github.com/fastly/terraform-provider-fastly/fastly/block_fastly_service_v1_requestsetting.go:122 +0
```

**3/ Quick & dirty solution**

It seems the tfstate file is read as a `int`, but the provider expects a `uint`. 
I cloned the repository on my local machine to change locally change the cast

In `terraform-provider-fastly/fastly/block_fastly_service_v1_requestsetting.go`, I changed:

```
		if v, ok := modified["max_stale_age"]; ok {
			// Old code: opts.MaxStaleAge = gofastly.Uint(v.(uint))
			helper := uint(v.(int))
			opts.MaxStaleAge = &helper
		}
```

This solved the issue for us.

--- a/fastly/block_fastly_service_v1_requestsetting.go
+++ b/fastly/block_fastly_service_v1_requestsetting.go
@@ -119,7 +119,7 @@ func (h *RequestSettingServiceAttributeHandler) Process(d *schema.ResourceData, 
 			opts.BypassBusyWait = gofastly.CBool(v.(bool))
 		}
 		if v, ok := modified["max_stale_age"]; ok {
-			opts.MaxStaleAge = gofastly.Uint(v.(uint))
+			opts.MaxStaleAge = gofastly.Uint(uint(v.(int)))
 		}
 		if v, ok := modified["hash_keys"]; ok {
 			opts.HashKeys = gofastly.String(v.(string))

--- a/fastly/block_fastly_service_v1_requestsetting_test.go
+++ b/fastly/block_fastly_service_v1_requestsetting_test.go
@@ -67,6 +67,13 @@ func TestAccFastlyServiceV1RequestSetting_basic(t *testing.T) {
 		XForwardedFor:    "append",
 		MaxStaleAge:      uint(90),
 	}
+	rq2 := gofastly.RequestSetting{
+		Name:             "alt_backend",
+		RequestCondition: "serve_alt_backend",
+		DefaultHost:      "tftestingother.tftesting.net.s3-website-us-west-2.amazonaws.com",
+		XForwardedFor:    "append",
+		MaxStaleAge:      uint(900),
+	}
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
@@ -74,10 +81,23 @@ func TestAccFastlyServiceV1RequestSetting_basic(t *testing.T) {
 		CheckDestroy:      testAccCheckServiceV1Destroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccServiceV1RequestSetting(name, domainName1),
+				Config: testAccServiceV1RequestSetting(name, domainName1, "90"),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
 					testAccCheckFastlyServiceV1RequestSettingsAttributes(&service, []*gofastly.RequestSetting{&rq1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "request_setting.#", "1"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "condition.#", "1"),
+				),
+			},
+			{
+				Config: testAccServiceV1RequestSetting(name, domainName1, "900"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1RequestSettingsAttributes(&service, []*gofastly.RequestSetting{&rq2}),
 					resource.TestCheckResourceAttr(
 						"fastly_service_v1.foo", "name", name),
 					resource.TestCheckResourceAttr(
@@ -134,7 +154,7 @@ func testAccCheckFastlyServiceV1RequestSettingsAttributes(service *gofastly.Serv
 	}
 }
 
-func testAccServiceV1RequestSetting(name, domain string) string {
+func testAccServiceV1RequestSetting(name, domain, maxStaleAge string) string {
 	return fmt.Sprintf(`
 resource "fastly_service_v1" "foo" {
   name = "%s"
@@ -167,11 +187,11 @@ resource "fastly_service_v1" "foo" {
     default_host      = "tftestingother.tftesting.net.s3-website-us-west-2.amazonaws.com"
     name              = "alt_backend"
     request_condition = "serve_alt_backend"
-    max_stale_age     = 90
+    max_stale_age     = %s
   }
 
   default_host = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
 
   force_destroy = true
-}`, name, domain)
+}`, name, domain, maxStaleAge)
 }

